From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 5 Jun 2020 17:02:35 +0200
Subject: backends/native: Drop users of clutter_input_device_set_stage()

There is no getter, so doing this is now pointless.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1486
---
 src/backends/native/meta-seat-native.c                 | 11 -----------
 src/backends/native/meta-virtual-input-device-native.c |  4 ----
 2 files changed, 15 deletions(-)

diff --git a/src/backends/native/meta-seat-native.c b/src/backends/native/meta-seat-native.c
index 413d101..9863863 100644
--- a/src/backends/native/meta-seat-native.c
+++ b/src/backends/native/meta-seat-native.c
@@ -1459,11 +1459,8 @@ evdev_add_device (MetaSeatNative         *seat,
 {
   ClutterInputDeviceType type;
   ClutterInputDevice *device, *logical = NULL;
-  ClutterActor *stage;
 
   device = meta_input_device_native_new (seat, libinput_device);
-  stage = CLUTTER_ACTOR (meta_seat_native_get_stage (seat));
-  _clutter_input_device_set_stage (device, CLUTTER_STAGE (stage));
 
   seat->devices = g_slist_prepend (seat->devices, device);
 
@@ -2474,7 +2471,6 @@ meta_seat_native_constructed (GObject *object)
 {
   MetaSeatNative *seat = META_SEAT_NATIVE (object);
   ClutterInputDevice *device;
-  ClutterStage *stage;
   MetaEventSource *source;
   struct udev *udev;
   struct xkb_keymap *xkb_keymap;
@@ -2482,8 +2478,6 @@ meta_seat_native_constructed (GObject *object)
   device = meta_input_device_native_new_virtual (
       seat, CLUTTER_POINTER_DEVICE,
       CLUTTER_INPUT_MODE_LOGICAL);
-  stage = meta_seat_native_get_stage (seat);
-  _clutter_input_device_set_stage (device, stage);
   seat->pointer_x = INITIAL_POINTER_X;
   seat->pointer_y = INITIAL_POINTER_Y;
   _clutter_input_device_set_coords (device, NULL,
@@ -2494,7 +2488,6 @@ meta_seat_native_constructed (GObject *object)
   device = meta_input_device_native_new_virtual (
       seat, CLUTTER_KEYBOARD_DEVICE,
       CLUTTER_INPUT_MODE_LOGICAL);
-  _clutter_input_device_set_stage (device, stage);
   seat->core_keyboard = device;
 
   udev = udev_new ();
@@ -2932,15 +2925,11 @@ meta_seat_native_set_stage (MetaSeatNative *seat,
     return;
 
   seat->stage = stage;
-  _clutter_input_device_set_stage (seat->core_pointer, stage);
-  _clutter_input_device_set_stage (seat->core_keyboard, stage);
 
   for (l = seat->devices; l; l = l->next)
     {
       ClutterInputDevice *device = l->data;
 
-      _clutter_input_device_set_stage (device, stage);
-
       if (clutter_input_device_get_device_mode (device) == CLUTTER_INPUT_MODE_PHYSICAL)
         {
           ClutterEvent *device_event;
diff --git a/src/backends/native/meta-virtual-input-device-native.c b/src/backends/native/meta-virtual-input-device-native.c
index a87e681..7200472 100644
--- a/src/backends/native/meta-virtual-input-device-native.c
+++ b/src/backends/native/meta-virtual-input-device-native.c
@@ -694,7 +694,6 @@ meta_virtual_input_device_native_constructed (GObject *object)
   MetaVirtualInputDeviceNative *virtual_evdev =
     META_VIRTUAL_INPUT_DEVICE_NATIVE (object);
   ClutterInputDeviceType device_type;
-  ClutterStage *stage;
 
   device_type = clutter_virtual_input_device_get_device_type (virtual_device);
 
@@ -707,9 +706,6 @@ meta_virtual_input_device_native_constructed (GObject *object)
                                           device_type,
                                           CLUTTER_INPUT_MODE_PHYSICAL);
 
-  stage = meta_seat_native_get_stage (virtual_evdev->seat);
-  _clutter_input_device_set_stage (virtual_evdev->device, stage);
-
   g_signal_emit_by_name (virtual_evdev->seat,
                          "device-added",
                          virtual_evdev->device);
